bpfd/src/bpf.rs

```diff
@@ -553,4 +561,163 @@ impl<'a> BpfManager<'a> {
         }
         results
     }
+
+    pub(crate) async fn process_commands(&mut self) {
+        loop {
+            // Start receiving messages
+            select! {
+                biased;
+                _ = shutdown_handler() => {
+                    info!("Signal received to stop command processing");
+                    break;
+                }
+                Some(cmd) = self.commands.recv() => {
+                    match cmd {
+                        Command::LoadXDP(args) => self.load_xdp_command(args).await.unwrap(),
+                        Command::LoadTC(args) => self.load_tc_command(args).await.unwrap(),
+                        Command::LoadTracepoint(args) => self.load_tracepoint_command(args).await.unwrap(),
+                        Command::Unload(args) => self.unload_command(args).await.unwrap(),
+                        Command::List { responder } => {
+                            let progs = self.list_programs();
+                            // Ignore errors as they'll be propagated to caller in the RPC status
+                            let _ = responder.send(progs);
+                        }
+                    }
+                }
+            }
+            info!("Stopping processing commands");
+        }
+    }
+
+    async fn load_xdp_command<'a>(&mut self, args: LoadXDPArgs) -> anyhow::Result<()> {
+        let res = if let Ok(if_index) = get_ifindex(&args.iface) {
+            let prog_data = ProgramData::new(
+                args.location,
+                args.section_name.clone(),
+                args.global_data,
+                args.username,
+            )
+            .await?;
+
+            let prog_result = Ok(Program::Xdp(XdpProgram {
+                data: prog_data.clone(),
+                info: XdpProgramInfo {
+                    if_index,
+                    current_position: None,
+                    metadata: command::Metadata {
+                        priority: args.priority,
+                        // This could have been overridden by image tags
+                        name: prog_data.section_name,
+                        attached: false,
+                    },
+                    proceed_on: args.proceed_on,
+                    if_name: args.iface,
+                },
+            }));
+
+            match prog_result {
+                Ok(prog) => self.add_program(prog, args.id).await,
+                Err(e) => Err(e),
```

Comment:
```suggestion
            let prog = Program::Xdp(XdpProgram {
                data: prog_data.clone(),
                info: XdpProgramInfo {
                    if_index,
                    current_position: None,
                    metadata: command::Metadata {
                        priority: args.priority,
                        // This could have been overridden by image tags
                        name: prog_data.section_name,
                        attached: false,
                    },
                    proceed_on: args.proceed_on,
                    if_name: args.iface,
                },
            });
            
            self.add_program(prog, args.id).await,
```

This should do it since prog_result can never be ERR as it is now, same for other functions.

Reply:
Good catch!